src/packbeam_api.erl

```diff
@@ -707,21 +707,43 @@ strip_padding(Data) ->
     Data.
 
 %% @private
-uncompress_literals(Chunks) ->
+maybe_uncompress_literals(Chunks) ->
     case proplists:get_value("LitT", Chunks) of
         undefined ->
             {Chunks, undefined};
-        <<_Header:4/binary, Data/binary>> ->
-            UncompressedData = zlib:uncompress(Data),
-            {
-                lists:keyreplace(
-                    "LitT",
-                    1,
-                    Chunks,
-                    {"LitU", UncompressedData}
-                ),
-                UncompressedData
-            }
+        LitT ->
+            <<_Header:4/binary, Data/binary>> = LitT,
```

Comment:
_Header actually is the size that is 0 when the literals are uncompressed.

So this can be simplified with proper matching.

```
<<0:32, Data/binary>> -> {Chunks, Data};
<<_Size:32, Data>> -> do_uncompress(Chunks, Data)
```

and you no longer need literals_need_uncompress.

Reply:
This was not exactly right, but put me on the right track. Thank you!